cms-adapter/src/Rsbc.Dmf.CaseManagement/Manager/CaseManager.cs

```diff
@@ -4184,40 +4184,111 @@ public DateTimeOffset GetDpsUploadedDate()
+            // Get the guid for the 
 
-            var mostRecentRecord = dynamicsContext.bcgov_documenturls
-                .Where(x => x.dfp_uploadeddate != null 
-                && x.dfp_submittalstatus == 100000010 // Document status Uplaoded
-                && x.dfp_dpsclass.Contains("9 - General") // Check for 9 - General
-                && x.dfp_DocumentTypeID.dfp_businessarea != 100000001) // Does not contain remedial
-                .OrderBy(i => i.dfp_uploadeddate) // order by oldest uploaded date 
-                .Take(1)
-                .FirstOrDefault();
+            Guid? eraResultsGuid = null;
 
-            if (mostRecentRecord != null)
+            var tempQuery = dynamicsContext.dfp_submittaltypes.Where(x => x.dfp_name == "ERA Results").FirstOrDefault();
+            if(tempQuery != null)
             {
-                return mostRecentRecord.dfp_uploadeddate.Value;
+                eraResultsGuid = tempQuery.dfp_submittaltypeid.Value;
             }
-            else
+
+
+            Guid? reExamResultsGuid = null;
+
+            tempQuery = dynamicsContext.dfp_submittaltypes.Where(x => x.dfp_name == "Road Test - ICBC Re-exam").FirstOrDefault();
+            if (tempQuery != null)
             {
-                return DateTimeOffset.UtcNow;
+                reExamResultsGuid = tempQuery.dfp_submittaltypeid.Value;
             }
+
+                var mostRecentRecord = dynamicsContext.bcgov_documenturls
+                    .Expand(x => x.dfp_DocumentTypeID)
+                    .Where(x => x.dfp_uploadeddate != null
+                    && x.dfp_submittalstatus == 100000010 // Document status Uplaoded
+                    && x.dfp_dpsclass.Contains("9 - General") // Check for 9 - General
+                    && x.dfp_DocumentTypeID.dfp_businessarea != 100000001 // Does not contain remedial
+                    && x._dfp_documenttypeid_value != eraResultsGuid
+                    && x._dfp_documenttypeid_value != reExamResultsGuid )// Exclude ERA Results and Road Test - ICBC Re-exam docuemnt types
+                    .OrderBy(i => i.dfp_uploadeddate) // order by oldest uploaded date 
+                    .Take(1)
+                    .FirstOrDefault();
+
+                if (mostRecentRecord != null)
+                {
+                    return mostRecentRecord.dfp_uploadeddate.Value;
+                }
+                else
+                {
+                    return DateTimeOffset.UtcNow;
+                }
+            
+
         }
 
+        /// <summary>
+        /// GetDpsProcessingDateForERAResults
+        /// </summary>
+        /// <returns></returns>
+        public DateTimeOffset GetDpsUplaodDateForERAResults()
+        {
+            Guid? eraResultsGuid = null;
+
+            var tempQuery = dynamicsContext.dfp_submittaltypes.Where(x => x.dfp_name == "ERA Results").FirstOrDefault();
```

Comment:
the next time we touch this code or add document queries, let's move the document related methods to DocumentManager, which already has submittaltype and bcgovdocumenturl queries